pkg/adapter/endpoint.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package adapter
+
+import (
+	apiv1alpha1 "sigs.k8s.io/external-dns/apis/v1alpha1"
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+func ToInternalEndpoint(crdEp *apiv1alpha1.Endpoint) *endpoint.Endpoint {
+	if crdEp == nil {
+		return nil
+	}
+	p := make(endpoint.ProviderSpecific)
+	for _, ps := range crdEp.ProviderSpecific {
+		p.Set(ps.Name, ps.Value)
+	}
+	ep := &endpoint.Endpoint{
```

Comment:
is `ep` required here, we could just return, and should not be &endpoint.Endpoint where possible ?